php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/PHPStanReportParser.java

```diff
@@ -184,6 +192,15 @@ private String getCurrentFile(String fileName) {
         }
         if (rootDirectory.isFolder()) {
             FileObject current = rootDirectory.getFileObject(sanitizedFileName);
+            if (current == null) {
+                // NETBANS-3022 try checking relative file path from working directory
+                if (workDir != null) {
+                    rootDirectory = FileUtil.toFileObject(workDir);
```

Comment:
Nitpick: Although cached, this is quite a time-consuming operation. Consider adding a class field

```java
@Nullable
private final FileObject workDir;
```

But this is really just a note, not a problem at all, of course.

Reply:
Does it mean that we should change the following to it?
https://github.com/apache/netbeans/blob/1ee4fdb02036bd3d626a1484181a9b9a7bc785e1/php/php.code.analysis/src/org/netbeans/modules/php/analysis/parsers/PHPStanReportParser.java#L59

e.g.
```java
    @Nullable
    private final FileObject workDir;
    private PHPStanReportParser(FileObject root, File workDir) throws SAXException {
        this.xmlReader = FileUtils.createXmlReader();
        this.root = root;
        this.workDir = FileUtil.toFileObject(workDir);
    }
```

Reply:
Or should return `FileObject` instead of `File` in findWorkDir() method?
https://github.com/apache/netbeans/blob/1ee4fdb02036bd3d626a1484181a9b9a7bc785e1/php/php.code.analysis/src/org/netbeans/modules/php/analysis/commands/PHPStan.java#L138-L149

Reply:
It is up to you to decide. I remember that personally, I was trying to use `FileObject` as much as possible because that is how NetBeans works with files. However, `FileObject` has many disadvantages too, e.g. if it is `null` you have no idea what the original file path was (typically problem in an API, here, `File` is typically a better choice).

Back to this situation - I would perhaps try to use `@Nullable FileObject ` for `findWorkDir` method (and log `ERROR` if `projectDir` is not found because that should never happen, right?).

Thanks!

Reply:
I'll use `FileObject` and leave the log if the project directory is not found. Thank you!